File: dalat-filter/models/place.py
from database import Base
from sqlalchemy import Boolean, Column, Integer, String, Float, Text
# from sqlalchemy.orm import relationship

class Place(Base):
    __tablename__ = "places"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    category_id = Column(Integer, nullable=False)
    address = Column(String, nullable=False)
    latitude = Column(Float, nullable=False)
    longitude = Column(Float, nullable=False)
    rating = Column(Float, nullable=False)
    description = Column(Text)
    adult_fare = Column(Integer, nullable=True)
    child_fare = Column(Integer, nullable=True)
    image_code = Column(String, nullable=True)
    other_images = Column(String, nullable=True)
    creator_id = Column(Integer, nullable=False)
    show = Column(Boolean, nullable=False)

File: dalat-be/src/main/java/com/dan/dalat/models/Place.java
package com.dan.dalat.models;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.time.LocalTime;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Entity
@Table(name = "places")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Place {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    String name;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id")
    @OnDelete(action = OnDeleteAction.SET_NULL)
    Category category;

    String address;
    BigDecimal latitude;
    BigDecimal longitude;
    Double rating;

    @Column(columnDefinition = "TEXT")
    String description;

    String imageCode;
    List<String> otherImages;

    @Column(name = "`show`")
    boolean show;

    Integer childFare;
    Integer adultFare;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "creator_id")
    User creator;

    LocalTime openTime;
    LocalTime closeTime;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "place_hobby",
            joinColumns = @JoinColumn(name = "place_id"),
            inverseJoinColumns = @JoinColumn(name = "hobby_id")
    )
    Set<Hobby> hobbies = new HashSet<>();

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "place_service",
            joinColumns = @JoinColumn(name = "place_id"),
            inverseJoinColumns = @JoinColumn(name = "service_id")
    )
    Set<TourismService> services = new HashSet<>();
}
